### ISSUE TITLE
Download error: "Unexpectedly found nil while implicitly unwrapping an Optional value"

### ISSUE THREAD

**Issue by unboxme** [dropbox/SwiftyDropbox#319]
Hi Dropbox team, 

It isn't possible to download file without `DownloadRequestFile` instance retain. Here is the force unwrap [crash line](https://github.com/dropbox/SwiftyDropbox/blob/6fb87816d35c7d7f3c2eac373ef1d0658af87ead/Source/SwiftyDropbox/Shared/Handwritten/DropboxTransportClient.swift#L182).

Code example:
```Swift
fileRoutes?.download(path: "/Apps/file.json", destination: { url, response in
    let documentsDirectory = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)[0]
    let destinationURL = documentsDirectory.appendingPathComponent("file.json")
    return destinationURL
})
```

Do you have a plans to fix it or is it a feature?

**Comment by greg-db** [comment]
Thanks for the report! This looks like a bug. We'll look into it.

**Comment by greg-db** [comment]
This should be fixed in the next release.

In the meantime, you can avoid this crash by including a `response` handler (even if you don't actually have it do anything), like this:
```
.response(completionHandler: { _, _ in
    // no-op
})
```

**Comment by unboxme** [comment]
Yup, I know. Thank you!
I just let you know about that bug — I don't know who else will try to use your SDK like that in real project :)

**Comment by greg-db** [comment]
This should be fixed in v7.0.1. Thanks!
